### PR TITLE
Allow both teams to release new smithy-rs versions independently

### PR DESCRIPTION
**Pull request by david-perez**
This commit modifies the `CODEOWNERS` to allow both
`@awslabs/rust-sdk-owners` and `@awslabs/smithy-rs-server` teams to
release new versions of smithy-rs independently, without requiring peer
approval from the other team.

Syntax and semantics of `CODEOWNERS`: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by jdisanti — APPROVED** [review]
I can't tell from the GitHub documentation if this will end up requiring both groups or one of the groups, but I guess we'll find out.

**Comment by david-perez** [comment]
> I can't tell from the GitHub documentation if this will end up requiring both groups or one of the groups, but I guess we'll find out.

Under [this section](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file), there's an example saying it's an OR.

```
# In this example, any change inside the `/scripts` directory
# will require approval from @doctocat or @octocat.
/scripts/ @doctocat @octocat
```

**Comment by david-perez** [comment]
Let's merge this after #1617.

**Pull request merged**
